Features/Context/FeatureContext.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Dunglas\AngularCsrfBundle\Features\Context;
+
+use Behat\Behat\Context\SnippetAcceptingContext;
+use Dunglas\AngularCsrfBundle\Features\Context\Fixtures\TestKernel;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+class FeatureContext implements SnippetAcceptingContext
+{
+    private $app;
+    private $lastResponse;
+    private $lastThrowException;
+
+    public function __construct()
+    {
+        $this->app = new TestKernel();
+        $this->app->boot();
+    }
+
+    /**
+     * @Given I send POST request to :arg1 with valid csrf header
+     */
+    public function iSendPostRequestToWithValidCsrfHeader($path)
+    {
+        $request = Request::create($path, 'POST');
+        $request->headers->add(array(
+            'X-XSRF-TOKEN' => $this->app->getContainer()->get('dunglas_angular_csrf.token_manager')->getToken()->getValue()
+        ));
+        $this->lastResponse = $this->app->handle($request);
+    }
+
+    /**
+     * @Given I send POST request to :arg1 with invalid csrf header
+     */
+    public function iSendPostRequestToWithInvalidCsrfHeader($path)
+    {
+        $request = Request::create($path, 'POST');
+        $request->headers->add(array(
+            'X-XSRF-TOKEN' => 'invalid_csrf_token'
+        ));
+
+        $this->lastResponse = $this->app->handle($request);
+
+    }
+
+    /**
+     * @Then the response code should be :arg1
+     */
+    public function theResponseShouldCodeBe($code)
+    {
+        if ($this->lastResponse->getStatusCode() !== (int) $code) {
```

Comment:
A `@Transform` method should better fit: http://docs.behat.org/en/latest/guides/2.definitions.html#step-argument-transformations